src/common/templates/_ark-saml.tpl

```diff
@@ -0,0 +1,91 @@
+{{- define "arkcase.saml.config" -}}
+  {{- $config := $ -}}
+  {{- $results := dict -}}
+  {{-
+    $required :=
+      list
+        "entityId"
+        "identityProviderUrl"
+  -}}
+
+  {{- $missing := list -}}
+  {{- range $key := $required -}}
+    {{- if (not (hasKey $config $key)) -}}
+      {{- $missing = append $missing $key -}}
+    {{- end -}}
+  {{- end -}}
+  {{- if $missing -}}
+    {{- fail (printf "SAML Configuration is missing the following keys: %s" $missing) -}}
+  {{- end -}}
+
+  {{- /* The config is valid! */ -}}
+  {{- $results = set $results "saml" $config -}}
+  {{- $results | toYaml -}}
+{{- end -}}
+
+{{- define "arkcase.saml.compute" -}}
+  {{- $ctx := $ -}}
+  {{- if not (include "arkcase.isRootContext" $ctx) -}}
+    {{- fail "The single parameter must be the root context ($ or .)" -}}
+  {{- end -}}
+
+  {{- $samlconfig := dict -}}
+
+  {{- /* TODO: When we add KeyCloak et al, we need to support both local and global configurations */ -}}
+  {{- /* This value must be a map with configs, or a true-false string */ -}}
+  {{- $conf := (include "arkcase.tools.conf" (dict "ctx" $ctx "value" "sso" "detailed" true) | fromYaml) -}}
+  {{- $saml := dict -}}
+  {{- if $conf.found -}}
+    {{- $conf := $conf.value -}}
+    {{- if and $conf $conf.enabled $conf.protocol $conf.saml (include "arkcase.toBoolean" $conf.enabled) (eq "saml" $conf.protocol) -}}
+      {{- $samlconfig = (include "arkcase.saml.config" $conf.saml) -}}
+    {{- end -}}
+  {{- end -}}
+  {{- $samlconfig -}}
```

Comment:
So you have an issue here when the `$samlconfig` value is not set b/c there's no SAML configuration - it will return a string version of the map, which is distinct from a YAML version of the map, which is what is expected to be returned.

My recommendation is to include the contents of the `arkcase.saml.config` template directly into this one (namely in line 41) so by the time you get here, the `$samlconfig` variable will always contain an actual object vs. sometimes-an object-and-sometimes-a-string, and thus you can use `toYaml` when returning the result.

This was a lesson that took me by surprise: the templates return STRINGS ... ALWAYS ... so if you want to return a complex object, you must return it as a _parseable string_ (i.e. YAML or JSON).